### PR TITLE
noto-fonts-emoji: unstable-2019-10-22 → unstable-2020-08-20; twitter-color-emoji: 12.1.5 → 13.0.1

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

Update Noto Color Emoji and Twitter Color Emoji to support Unicode 13.0 for new emojis like 🏳️‍⚧️.

Following renderings are with the updated fonts from this PR:

* [Twitter Color Emoji](https://sterni.lv/tmp/twemoji.png) (the hole stems from the fact as far as I can see that twemoji is missing man with veil, person with veil and their variations)
* [Noto Color Emoji](https://sterni.lv/tmp/noto.png)

###### TODO

- [x] update `noto-fonts-emoji`
- [x] update / fix `twitter-emoji-color`
- [ ] test packages depending on these changes (aka `nixpkgs-review`)?

Also feedback especially on the new python packages would be appreciated. I'm not sure if there's a better way to handle the `extras` than I'm doing right now.

Also sorry for the huge PR, but all these changes are more or less related to each other, so opening thousands of PRs would be more of a hassle, I guess.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
Weird test failure for `fonttools` on ofborg, can anybody reproduce?

**Comment by sternenseemann** [comment]
Test failure on ofborg is very strange, I can reproduce it locally, but not 100% of the time. Seems to be related to timestamps, maybe nix is messing with it?

File: pkgs/data/fonts/noto-fonts/tools.nix [hunk 1]
```diff
@@ -1,30 +1,58 @@
...
-      sha256 = "0bn0rlbddxicw0h1dnl0cibgj6xjalja2qcm563y7kk3z5cdwhgq";
-    })
-  ];
-
   postPatch = ''
-    sed -ie "s^join(_DATA_DIR_PATH,^join(\"$out/third_party/ucd\",^" nototools/unicode_data.py
+    sed -i 's/use_scm_version=.*,/version="${version}",/' setup.py
   '';
 
+  buildInputs = [ setuptools_scm ];
```
Comment (jonringer) [inline pkgs/data/fonts/noto-fonts/tools.nix:25]:
```suggestion
  nativeBuildInputs = [ setuptools_scm ];
```

File: pkgs/development/python-modules/afdko/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  version = "3.5.0";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0wid4l70bxm297xgayyrgw5glhp6n92gh4sz1nd4rncgf1ziz8ck";
+  };
+
+  buildInputs = [
```
Comment (jonringer) [inline pkgs/development/python-modules/afdko/default.nix:20]:
```suggestion
  nativeBuildInputs = [
```

File: pkgs/development/python-modules/afdko/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    mutatorMath
+    ufoProcessor
+    ufonormalizer
+    psautohint
+  ];
+
+  checkPhase = ''
+    PATH="$PATH:$out/bin" py.test
+  '';
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/afdko/default.nix:45]: default for python packages
```suggestion
```

File: pkgs/development/python-modules/booleanOperations/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "booleanOperations";
+  version = "0.9.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1f41lb19m8azchl1aqz6j5ycbspb8jsf1cnn42hlydxd68f85ylc";
+    extension = "zip";
+  };
+
+  buildInputs = [ setuptools_scm ];
```
Comment (jonringer) [inline pkgs/development/python-modules/booleanOperations/default.nix:16]:
```suggestion
  nativeBuildInputs = [ setuptools_scm ];
```

File: pkgs/development/python-modules/booleanOperations/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  propagatedBuildInputs = [
+    fonttools
+    fs
+    pyclipper
+    defcon
+    fontPens
+  ];
+
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/booleanOperations/default.nix:26]: default
```suggestion
```

File: pkgs/development/python-modules/defcon/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    inherit pname version;
+    sha256 = "1lfqsvxmq1j0nvp26gidnqkj1dyxv7jalc6i7fz1r3nc7niflrqr";
+    extension = "zip";
+  };
+
+  propagatedBuildInputs = [
+    fonttools
+  ];
+
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/defcon/default.nix:20]:
```suggestion
```

File: pkgs/development/python-modules/fontMath/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "09xdqdjyjlx5k9ymi36d7hkgvn55zzjzd65l2yqidkfazlmh14ss";
+    extension = "zip";
+  };
+
+  propagatedBuildInputs = [ fonttools ];
+
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/fontMath/default.nix:18]:
```suggestion
```

File: pkgs/development/python-modules/fontParts/default.nix [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    unicodedata2  # fonttools[unicode] extra
+    lxml          # fonttools[lxml] extra
+    fs            # fonttools[ufo] extra
+    defcon
+    fontPens      # defcon[pens] extra
+    fontMath
+  ];
+
+  checkPhase = "python Lib/fontParts/fontshell/test.py";
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/fontParts/default.nix:29]:
```suggestion
```

File: pkgs/development/python-modules/fontParts/default.nix [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    fonttools
+    unicodedata2  # fonttools[unicode] extra
+    lxml          # fonttools[lxml] extra
+    fs            # fonttools[ufo] extra
+    defcon
+    fontPens      # defcon[pens] extra
+    fontMath
+  ];
+
+  checkPhase = "python Lib/fontParts/fontshell/test.py";
```
Comment (jonringer) [inline pkgs/development/python-modules/fontParts/default.nix:28]: to support alternate interpreters
```suggestion
  checkPhase = ''
    ${python.interpreter} Lib/fontParts/fontshell/test.py
  '';
```

File: pkgs/development/python-modules/mutatorMath/default.nix [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0r1qq45np49x14zz1zwkaayqrn7m8dn2jlipjldg2ihnmpzw29w1";
+    extension = "zip";
+  };
+
+  propagatedBuildInputs = [ fontMath unicodedata2 defcon ];
+
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/mutatorMath/default.nix:18]:
```suggestion
```

File: pkgs/development/python-modules/psautohint/default.nix [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    echo '#define PSAUTOHINT_VERSION "${version}"' > libpsautohint/src/version.h
+    sed -i '/use_scm_version/,+3d' setup.py
+    sed -i '/setup(/a \     version="${version}",' setup.py
+  '';
+
+  nativeBuildInputs = [ setuptools_scm ];
+
+  propagatedBuildInputs = [ fonttools lxml fs ];
+
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/psautohint/default.nix:31]:
```suggestion
```

File: pkgs/development/python-modules/pytest-randomly/default.nix [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  disabled = isPy27;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0s9cx692cdchfrjqx7fgf9wnm3fdac211a4hjq1cx9qqnbpdpl2z";
+  };
+
+  propagatedBuildInputs = [ numpy factory_boy faker ];
+
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/pytest-randomly/default.nix:19]:
```suggestion
```

File: pkgs/development/python-modules/ufoProcessor/default.nix [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+  pname = "ufoProcessor";
+  version = "1.9.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0ns11aamgavgsfj8qf5kq7dvzmgl0mhr1cbych2f075ipfdvva5s";
+    extension = "zip";
+  };
+
+  buildInputs = [ setuptools_scm ];
```
Comment (jonringer) [inline pkgs/development/python-modules/ufoProcessor/default.nix:17]:
```suggestion
  nativeBuildInputs = [ setuptools_scm ];
```

File: pkgs/development/python-modules/ufoProcessor/default.nix [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+    lxml
+    fonttools
+    fs
+    fontMath
+    fontParts
+    mutatorMath
+  ];
+
+  doCheck = true;
+
```
Comment (jonringer) [inline pkgs/development/python-modules/ufoProcessor/default.nix:30]:
```suggestion
```

File: pkgs/development/python-modules/ufonormalizer/default.nix [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.4.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0nv80x7l7sya5wzyfk9ss93r6bjzjljpkw4k8gibxp1rqrzkdms4";
+    extension = "zip";
+  };
+
+  doCheck = true;
+
```
Comment (jonringer) [inline pkgs/development/python-modules/ufonormalizer/default.nix:15]:
```suggestion
```

File: pkgs/development/python-modules/unicodedata2/default.nix [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub {
+    owner  = "mikekap";
+    repo   = pname;
+    rev    = version;
+    sha256 = "0p9brbiwyg98q52y0gfyps52xv57fwqfpq0mn18p1xc1imip3h2b";
+  };
+
+  checkInputs = [ pytest ];
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/unicodedata2/default.nix:15]:
```suggestion
  checkInputs = [ pytest ];
```

File: pkgs/development/python-modules/zopfli/default.nix [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.1.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0smaxh7iihjr9mwxw1ifc9vnlh3ra8l060dd1gbvp1963k0r68pd";
+    extension = "zip";
+  };
+
+  checkInputs = [ pytest ];
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/zopfli/default.nix:14]:
```suggestion
```

**Review by jonringer — COMMENTED** [review]
buildPythonPackage by default has `doCheck = true;`

`setuptools_scm` should be in nativeBuildInputs to avoid it being in the runtime closure

otherwise LGTM

File: pkgs/development/python-modules/fonttools/default.nix [hunk 18]
```diff
@@ -1,38 +1,66 @@
...
+    matplotlib
+    # pens
+    reportlab
   ];
 
   preCheck = ''
     export LC_ALL="en_US.UTF-8"
   '';
 
+  checkPhase = "pytest Tests fontTools";
```
Comment (jonringer) [inline pkgs/development/python-modules/fonttools/default.nix:62]:
```suggestion
  # avoid timing issues with timestamps
  checkPhase = ''
    pytest Tests fontTools -k 'not ttcompile_timestamp_calcs'
  '';
```

**Comment by sternenseemann** [comment]
@grahamcofborg build python3Packages.fonttools

File: pkgs/development/python-modules/fontParts/default.nix (hunk 9 above)
Comment (sternenseemann) [reply pkgs/development/python-modules/fontParts/default.nix]: Thanks, fixed!

File: pkgs/development/python-modules/fonttools/default.nix (hunk 18 above)
Comment (sternenseemann) [reply pkgs/development/python-modules/fonttools/default.nix]: Thanks, found two other tests with the same issue, should be good now.

**Comment by sternenseemann** [comment]
@jonringer Thanks for the good suggestions, let me know if I missed fixing any of them.

**Comment by jonringer** [comment]
cc @FRidh this is at 539 packages changed, around 1/4 of which are python packages. Should this go into staging?

```
$ nix-env -f /home/jon/.cache/nixpkgs-review/pr-96315/nixpkgs -qaP --xml --out-path --show-trace --meta
29 packages added:
python27Packages.ufonormalizer (init at 0.4.1) python27Packages.unicodedata2 (init at 13.0.0-2) python27Packages.zopfli (init at 0.1.6) python37Packages.afdko (init at 3.5.0) python37Packages.booleanOperations (init at 0.9.0) python37Packages.defcon (init at 0.7.2) python37Packages.fontMath (init at 0.6.0) python37Packages.fontParts (init at 0.9.2) python37Packages.fontPens (init at 0.2.4) python37Packages.mutatorMath (init at 3.0.1) python37Packages.psautohint (init at 2.1.0) python37Packages.pytest-randomly (init at 3.4.1) python37Packages.ufonormalizer (init at 0.4.1) python37Packages.ufoProcessor (init at 1.9.0) python37Packages.unicodedata2 (init at 13.0.0-2) python37Packages.zopfli (init at 0.1.6) python38Packages.afdko (init at 3.5.0) python38Packages.booleanOperations (init at 0.9.0) python38Packages.defcon (init at 0.7.2) python38Packages.fontMath (init at 0.6.0) python38Packages.fontParts (init at 0.9.2) python38Packages.fontPens (init at 0.2.4) python38Packages.mutatorMath (init at 3.0.1) python38Packages.psautohint (init at 2.1.0) python38Packages.pytest-randomly (init at 3.4.1) python38Packages.ufonormalizer (init at 0.4.1) python38Packages.ufoProcessor (init at 1.9.0) python38Packages.unicodedata2 (init at 13.0.0-2) python38Packages.zopfli (init at 0.1.6)

510 packages updated:
abcl adapta-gtk-theme adoptopenjdk-icedtea-web adoptopenjdk-icedtea-web adoptopenjdk-icedtea-web alchemy alda aldor alloy4 alloy5 alloy5 ammonite ammonite ammonite antlr antlr antlr antlr apache-directory-studio apache-flex-sdk apache-flex-sdk apache-jena apache-jena-fuseki apache-jena-fuseki apache-jena-fuseki apache-kafka apache-kafka apache-kafka apache-kafka apache-kafka apache-kafka apache-kafka apache-kafka apache-kafka apache-kafka apache-maven apache-storm ape ape-clex apktool arduino arduino-core arduino-core areca asciidoc-full asciidoc-full asciidoc-full-with-plugins asciidoctorj aspectj astrolabe-generator atlassian-cli avian avro-tools aws-mturk-clt axis2 axoloti azureus babashka baresip basex bazel bazel bazel bazel bazel-remote bazel-watcher belle-sip-unstable bfg-repo-cleaner bftools bitwig-studio bitwig-studio bitwig-studio bloop boot briss btrbk buck burpsuite cassandra cassandra cassandra cassandra ccache ccache-links-wrapper- ccemux cfr charles charles checkstyle chrome-gnome-shell chromium chromium-beta chromium-dev clevis clj-kondo clojure-lsp clooj closure-compiler collectd collectd-data confluent-platform coursier cvc4 davmail dbus-java dbvisualizer dex2jar DisnixWebService disorderfs ditaa dotty drip dwarf-fortress-full ec2-api-tools eclipse-scala-sdk elasticmq elasticmq-server elementary-session-settings-unstable emem emojione emscripten emscripten-libxml2 emscripten-xmlmirror-unstable epubcheck ergo erlang erlang exhibitor flashtool flexibee flink flink fop foundationdb foundationdb foundationdb foundationdb freemind freenet-build01475 freetts frostwire frostwire-desktop FurnitureLibraryEditor galen gama ganttproject-bin gcs geogebra GeoIPJava gephi global-platform-pro gnome-control-center gnome-session gnome-shell gnome-shell-extension-night-theme-switcher gnome-shell-gsconnect gnome-shell-pomodoro gnome-terminal gnome-tweaks gogui google-chrome google-chrome-beta google-chrome-dev graal gradle gradle gradle gremlin-console groove groovy gvisor h2 hadoop hadoop hdf-java hdfview herbstluftwm hol4-k.10 hsqldb html5validator hxjava hydra-ant-logger i2p ili2c imagej impressive ino ipscan jabref jadx jameica jasmin java-cup java-service-wrapper jbake jbidwatcher jd-gui jdepend jdiskreport jedit jflex jing-trang jing-trang jitsi jmeter jmol jmx-prometheus-httpserver jnetmap jogl jruby json-c jugglinglab junit junit junixsocket jvmci jython keycloak kodi kodi kodi kodi-platform kodi-plugin-inputstream-adaptive kodi-plugin-peripheral.joystick kodi-plugin-peripheral.steamcontroller kodi-plugin-pvr-hdhomerun kodi-plugin-pvr-hts kodi-plugin-pvr-iptvsimple kodi-plugin-vfs.libarchive kodi-plugin-vfs.sftp kodi-wayland kodi-wayland-with-plugins kodi-with-plugins kodi-with-plugins kodi-with-plugins kotlin ktlint LanguageTool legends-browser leiningen leo3 liberation-fonts liberation-fonts liberation-sans-narrow liberation-sans-narrow liblinphone-unstable libmatthew-java libreoffice libreoffice libreoffice libreoffice libreoffice libreoffice lime linphone-desktop-unstable liquibase logisim logstash logstash logstash-oss logstash-oss luksmeta lumo ma1sd macse martyr marvin maven-hello maven-hello mercury metabase metals micronaut mill minc-widgets mindustry mindustry minecraft-launcher mkgmap mkrom-unstable monosat msgviewer mucommander multimc munin mxisd mysql-workbench mysql-workbench nailgun nco neo4j netlogo notmuch-bower noto-fonts-emoji (2019-10-22 → 2020-08-20) ocaml4.10.0-ocaml-r ocaml4.10.0-phylogenetics-unstable octave octave octave odpdown omegat opa open-ecard openapi-generator-cli openapi-generator-cli openfire opengrok openjdk openjdk openjump openmodelica opentsdb pakcs papermc patroni pdf2htmlEX pdftk perl5.28.3-Inline-Java perl5.28.3-Net-ZooKeeper perl5.30.3-awstats perl5.30.3-gscan2pdf perl5.30.3-Inline-Java perl5.30.3-Net-ZooKeeper picard-tools picoLisp plantuml plm pmd polymake processing processing projectlibre python2.7-foundationdb python2.7-foundationdb python2.7-foundationdb python2.7-foundationdb python2.7-kazoo python2.7-PyStringTemplate python2.7-rpy2 python2.7-sphinxcontrib-plantuml python2.7-zake python3.7-arviz python3.7-baselines python3.7-dm-sonnet python3.7-edward python37Packages.fonttools (4.11.0 → 4.14.0) python3.7-foundationdb python3.7-foundationdb python3.7-foundationdb python3.7-foundationdb python3.7-graph_nets python3.7-kazoo python3.7-mask-rcnn python3.7-monosat python37Packages.nototools (2019-10-21 → 0.2.12) python3.7-optuna python3.7-pymc3 python3.7-rl-coach python3.7-rpy2 python3.7-skein python3.7-slither-analyzer python3.7-sphinxcontrib-plantuml python3.7-tensorflow python3.7-tensorflow python3.7-tensorflow python3.7-tensorflow-gpu python3.7-tensorflow_probability python3.7-tflearn python3.7-zake python38Packages.fonttools (4.11.0 → 4.14.0) python3.8-foundationdb python3.8-foundationdb python3.8-foundationdb python3.8-foundationdb python3.8-kazoo python3.8-monosat python38Packages.nototools (2019-10-21 → 0.2.12) python3.8-rpy2 python3.8-skein python3.8-slither-analyzer python3.8-sphinxcontrib-plantuml python3.8-zake quantomatic R R rabbitmq-java-client rascal rhino rictydiminished-with-firacode riemann ripcord RStudio RStudio sage sage sage saxon saxon-he saxonb saxonb sbt sbt sbt-extras scala scala scala scala scalafix scalafmt scfbuild schemaspy scylladb selendroid-standalone selenium-server-standalone seyren shattered-pixel-dungeon signal-cli slither-analyzer smc snpeff soapui solc solr spring spring-boot-cli spring-boot-cli springlobby sqlitebrowser squirrel-sql srcml sshoogr stdenv-linux stm32cubemx subgit subsonic swagger-codegen sweethome3d-application swi-prolog swi-prolog swingsane swt tabula tang teensyduino tessera TexturesLibraryEditor tlaplus tomcat-connectors tomcat-native tree-sitter tsm-client tuxguitar tvbrowser-bin twemoji-color-font twitter-color-emoji (12.1.5 → 13.0.1) udiskie umlet ums ungoogled-chromium unoconv varscan VASSAL vimplugin-vim-grammarous virtualbox virtualbox virtualbox virtualbox virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules virtualbox-modules visidata visualvm vue vuze webbrowser weka welkin whirlpool-gui workcraft xits-math xmage xmind xmloscopy XtreemFS yEd youtrack yuicompressor zap zgrviewer zipkin-server zk-shell zkfuse zlib zookeeper zookeeper_mt zxing
```

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build python3Packages.fonttools python3Packages.afdko

**Comment by sternenseemann** [comment]
Now `afdko` tests fail on ofborg which work for me.

**Comment by sternenseemann** [comment]
`afdko` tests where broken on `aarch64`, from a look at their bug tracker it seems to me that `makeotf` and `tx` only work properly on `x86_64` (i. e. tests fail at least on `aarch64` and on `i586`). I've restricted test running in c386ecb952df659550c753103f355f63699aa72b.

Not sure if we should also mark the package as broken on those platforms. At least we only need to build the fonts on one platform…

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build python3Packages.fonttools python3Packages.afdko python3Packages.nototools

File: pkgs/data/fonts/noto-fonts/default.nix [hunk 19]
```diff
@@ -110,25 +111,29 @@ in
...
-      sha256 = "0qmnnjpp5lza6g5m3ki6hj46p891h9vl42k3acd0qw8i0jj5yn2c";
+      rev = "1bc491419fa2925d018f27bfe702792031be0e68";
+      sha256 = "1vak4s1p4wlwzpnqfb1c2sg62q82gnjpnmqrfz8xl6bd0z55imzy";
     };
 
-    buildInputs = [ cairo ];
-    nativeBuildInputs = [ pngquant optipng which cairo pkgconfig imagemagick ]
-                     ++ (with python3Packages; [ python fonttools nototools ]);
+    nativeBuildInputs = [ cairo imagemagick zopfli pngquant which
+                          pkgconfig emojiPythonEnv ];
```
Comment (jonringer) [inline pkgs/data/fonts/noto-fonts/default.nix:129]: most common formatting with multiline lists
```suggestion
    nativeBuildInputs = [
      cairo
      imagemagick
      zopfli
      pngquant
      which
      pkgconfig
      emojiPythonEnv
    ];
```

File: pkgs/development/python-modules/fontPens/default.nix [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1za15dzsnymq6d9x7xdfqwgw4a3003wj75fn2crhyidkfd2s3nd6";
+    extension = "zip";
+  };
+
+  propagatedBuildInputs = [ fonttools ];
+
+  # can't check due to circular dependency with fontParts
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/fontPens/default.nix:16]: If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

File: pkgs/top-level/python-packages.nix [hunk 21]
```diff
@@ -4932,6 +4950,8 @@ in {
     else
       callPackage ../development/python-modules/mutagen { };
 
+  mutatorMath = callPackage ../development/python-modules/mutatorMath { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:4953]: Please normalize the attribute name. According to the [python contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines), package names within nix should follow [PEP 0503](https://www.python.org/dev/peps/pep-0503/#normalized-names)

File: pkgs/top-level/python-packages.nix [hunk 22]
```diff
@@ -4136,6 +4148,12 @@ in {
     inherit python;
   }));
 
+  fontPens = callPackage ../development/python-modules/fontPens { };
+
+  fontMath = callPackage ../development/python-modules/fontMath { };
+
+  fontParts = callPackage ../development/python-modules/fontParts { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:4155]: Please normalize the attribute name. According to the [python contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines), package names within nix should follow [PEP 0503](https://www.python.org/dev/peps/pep-0503/#normalized-names)

File: pkgs/top-level/python-packages.nix [hunk 23]
```diff
@@ -2087,6 +2093,8 @@ in {
 
   boltztrap2 = callPackage ../development/python-modules/boltztrap2 { };
 
+  booleanOperations = callPackage ../development/python-modules/booleanOperations { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:2096]: Please normalize the attribute name. According to the [python contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines), package names within nix should follow [PEP 0503](https://www.python.org/dev/peps/pep-0503/#normalized-names)

File: pkgs/top-level/python-packages.nix [hunk 24]
```diff
@@ -1719,6 +1721,10 @@ in {
     inherit (pkgs.darwin.apple_sdk.frameworks) CFNetwork Security;
   };
 
+  ufonormalizer = callPackage ../development/python-modules/ufonormalizer { };
+
+  ufoProcessor = callPackage ../development/python-modules/ufoProcessor { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:1726]: Please normalize the attribute name. According to the [python contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines), package names within nix should follow [PEP 0503](https://www.python.org/dev/peps/pep-0503/#normalized-names)

File: pkgs/top-level/python-packages.nix (hunk 21 above)
Comment (sternenseemann) [reply pkgs/top-level/python-packages.nix:4953]: What about the `name` attribute? Should it be like the attribute name or like in the pypi archive (so `inherit pname` is possible)?

File: pkgs/development/python-modules/fontPens/default.nix (hunk 20 above)
Comment (sternenseemann) [reply pkgs/development/python-modules/fontPens/default.nix]: Added in 21d369c55f5e3952cc372764b0f58b778baac62c.

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build python3Packages.fonttools python3Packages.afdko python3Packages.nototools

File: pkgs/data/fonts/noto-fonts/default.nix (hunk 19 above)
Comment (sternenseemann) [reply pkgs/data/fonts/noto-fonts/default.nix]: Done in 84dfeee7d823bc2e0744b703681ae891526cd0e6.

File: pkgs/top-level/python-packages.nix (hunk 21 above)
Comment (jonringer) [reply pkgs/top-level/python-packages.nix]: pname is fine to keep with the irregular format, so that it agrees with the fetchpypi usage

Comment (jonringer) [reply pkgs/top-level/python-packages.nix]: the main  thing is that referencing the package should be done through the normalized name (e.g. `propagatedBuildInputs = [ mutatormath ];`

Comment (sternenseemann) [reply pkgs/top-level/python-packages.nix]: Then, all should be well now.

**Review by jonringer — APPROVED** [review]
diff LGTM

built the first ~500 or so of the packages, and the failures don't look related to python package bumps

**Pull request merged**